xarray/core/variable.py

```diff
@@ -529,12 +529,14 @@ def _dask_finalize(self, results, array_func, *args, **kwargs):
         return Variable(self._dims, data, attrs=self._attrs, encoding=self._encoding)
 
     @property
-    def values(self):
+    def values(self) -> np.ndarray:
         """The variable's data as a numpy.ndarray"""
-        return _as_array_or_item(self._data)
+        return _as_array_or_item(
+            self._data
+        )  # type:ignore[return-value]  # TODO: fix this mess
```

Comment:
+1

See these comments for context:
https://github.com/pydata/xarray/blob/073f414eb810af02e2ad12d35bcd313558921837/xarray/core/variable.py#L257-L262

https://github.com/numpy/numpy/issues/4337 is actually fixed now, so perhaps we could indeed simply use `np.asarray` here?

Reply:
The second issue is still open.
But to me it seems that everywhere it is expected that `Variable.values` returns a `np.ndarray`, so I am not sure where this special casing is required.
Is there a test where this special case can be seen?